### DISCUSSION TITLE
This is not implemented, as it is not possible to implement Argon2i with acceptable performance in pure-PHP

### DISCUSSION THREAD

**Discussion by bybilgili** [joomla/joomla-cms#32444] [Q&A]
This is not implemented, as it is not possible to implement Argon2i with acceptable performance in pure-PHP

hello ; how can i fix this

**Comment by richard67** [comment]
See https://github.com/joomla/joomla-cms/issues/27375 

> You need to install `ext/sodium` through PECL or compile PHP with Argon support enabled (both of which require `libsodium` installed).
> 
> The `sodium_compat` library cannot polyfill all functions in pure PHP, you have hit one of those cases.

@bybilgili Does that help you? Please report back so we know if we can close this issue, or close it yourself if the above hint helped you to msolve it. Thanks in advance.

**Reply by josiahisaack** [reply]
ON Your Xampp control panel click on the first config TAB.
Open PHP (php.ini) on your notepad
Ctrl+F and seach for sodium
Enable the extension bearing the name sodium by DELETING the SEMICOLON at the beginning.
Save
Restart Apache by Stopping and Starting Again.
Then try to RUN C:\xampp\htdocs\magento2>
php bin/magento setup:install --base-url=“http://localhost/magento2/” --db-host=“localhost” --db-name=“magento2” --db-user=“root” --db-password=“” --admin-firstname=“admin” --admin-lastname=“admin” --admin-email=“user@example.com” --admin-user=“admin” --admin-password=“admin123” --language=“en_US” --currency=“USD” --timezone=“America/Chicago” --use-rewrites=“1” --backend-frontname=“admin” --search-engine=elasticsearch7 --elasticsearch-host=“localhost” --elasticsearch-port=9200

Goodluck.

**Reply by wilsonge** [reply]
This will work for magento - not joomla :)

**Comment by RickyThakur** [comment]
If you are using LAMPP then [this answer](https://magento.stackexchange.com/questions/331991/this-is-not-implemented-as-it-is-not-possible-to-implement-argon2i-with-accepta#answer-332273) can help.

**Comment by thecoachsmb** [comment]
There are **4 solutions** to resolve this error on **WINDOWS XAMPP**.

**Solution 1:-**

One of the reason for this error is sodium extension not enabled in php.ini file, so we need to enable it and re-run install command in the command prompt.

**To enable:-**

Go to C:\xampp\php\php.ini, find "extension=sodium" in it and then remove semicolon from beginning of this line to enable PHP extensions sodium. Save the file.

Then restart Apache means stop and start Apache from the Xampp Control Panel.

That’s it.

If you want to see[ more solutions, click here](https://www.thecoachsmb.com/troubleshoot-generally-occured-errors-in-magento-2-installation/).
